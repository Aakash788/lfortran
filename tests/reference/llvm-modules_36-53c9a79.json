{
    "basename": "llvm-modules_36-53c9a79",
    "cmd": "lfortran --no-color --show-llvm {infile} -o {outfile}",
    "infile": "tests/../integration_tests/modules_36.f90",
    "infile_hash": "a3b49d1f064fafdacd255475ac959ba94ab51101c75acda295c35118",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "llvm-modules_36-53c9a79.stdout",
    "stdout_hash": "820b4913da9849699664c4fcbfa25f95d8f2981097d0a009f1c99d2b",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}