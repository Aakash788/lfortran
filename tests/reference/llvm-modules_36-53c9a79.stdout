; ModuleID = 'LFortran'
source_filename = "LFortran"

%fpm_run_settings_polymorphic = type { i64, %fpm_run_settings* }
%fpm_run_settings = type { %fpm_build_settings, [2 x i8*], i8*, i8*, i1 }
%fpm_build_settings = type { i1 }
%__vtab_fpm_run_settings = type { i64 }

define void @__module_modules_36_fpm_main_01_cmd_run(%fpm_run_settings_polymorphic* %settings, i1* %test) {
.entry:
  %0 = alloca %__vtab_fpm_run_settings, align 8
  %1 = getelementptr %__vtab_fpm_run_settings, %__vtab_fpm_run_settings* %0, i32 0, i32 0
  store i64 0, i64* %1, align 4
  br label %return

return:                                           ; preds = %.entry
  ret void
}

define i32 @main(i32 %0, i8** %1) {
.entry:
  %call_arg_value = alloca i1, align 1
  call void @_lpython_set_argv(i32 %0, i8** %1)
  %settings = alloca %fpm_run_settings, align 8
  %2 = getelementptr %fpm_run_settings, %fpm_run_settings* %settings, i32 0, i32 2
  %3 = getelementptr %fpm_run_settings, %fpm_run_settings* %settings, i32 0, i32 4
  %4 = getelementptr %fpm_run_settings, %fpm_run_settings* %settings, i32 0, i32 1
  %5 = getelementptr %fpm_run_settings, %fpm_run_settings* %settings, i32 0, i32 3
  %6 = alloca %fpm_run_settings_polymorphic, align 8
  %7 = getelementptr %fpm_run_settings_polymorphic, %fpm_run_settings_polymorphic* %6, i32 0, i32 0
  store i64 0, i64* %7, align 4
  %8 = getelementptr %fpm_run_settings_polymorphic, %fpm_run_settings_polymorphic* %6, i32 0, i32 1
  store %fpm_run_settings* %settings, %fpm_run_settings** %8, align 8
  store i1 true, i1* %call_arg_value, align 1
  call void @__module_modules_36_fpm_main_01_cmd_run(%fpm_run_settings_polymorphic* %6, i1* %call_arg_value)
  ret i32 0
}

declare void @_lpython_set_argv(i32, i8**)
